src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs

```diff
@@ -8,13 +8,14 @@ namespace NServiceBus
 
     class MainPipelineExecutor : IPipelineExecutor
     {
-        public MainPipelineExecutor(IBuilder rootBuilder, IPipelineCache pipelineCache, MessageOperations messageOperations, INotificationSubscriptions<ReceivePipelineCompleted> receivePipelineNotification, Pipeline<ITransportReceiveContext> receivePipeline)
+        public MainPipelineExecutor(IBuilder rootBuilder, IPipelineCache pipelineCache, MessageOperations messageOperations, INotificationSubscriptions<ReceivePipelineCompleted> receivePipelineNotification, Pipeline<ITransportReceiveContext> receivePipeline, ConsecutiveFailuresCircuitBreaker consecutiveFailuresCircuitBreaker)
```

Comment:
We would never rate limit satellites?

Also is there a way of somehow encapsulate things better so that leaking the rate limiting into things is not necessary or at least less intrusive? For example having a dedicated PipelineExecutor that does it (or an executor decorator)